message.go

```diff
@@ -145,16 +145,33 @@ var zeroMessage = &Message{}
 
 // Concurrent type that can be safely shared between goroutines
 type messages struct {
-	sync.Mutex
+	sync.RWMutex
 	items []*Message
 }
 
 // messages methods
 
-// Addes new message pointer into concurrent messages slice
-func (messages *messages) append(item *Message) {
+// Adds new message pointer into concurrent messages slice
+func (messages *messages) Append(item *Message) {
```

Comment:
Sorry, could you describe for what we need to have an exported methods `Append`, `Copy`. It will be used under the hood only, no? Maybe I've forgotten something...

Reply:
`messages` itself is not exported, so having exported methods on it makes it clearer what other code (but only inside this package) could call from outside and which code (lock-less copy) it should not.

Reply:
@mitar When you "high level" read code of the library it can bring more mess for understanding. For me exported fields/methods are public interfaces always, and unexported are always private. I have got your point about focusing on the locks, but, maybe it is better to follow this convention (I have used it through all the project codebase)?

Reply:
Sure. Should I then name `copy` something like `copyInternal`?

Reply:
@mitar Of course!

Reply:
Done.